### ISSUE TITLE
[question] Why did you merge checks for vendor specific property prefix and experimental property?

### ISSUE THREAD

**Issue by aepfli** [racodond/sonar-css-plugin#122]
Hi!

Well first of all, nice plugin - pretty useful, and glad there is finally an scss plugin 

We are quite happy with the functionality, and we are using it, and got really dissapointed in our scss quality (project is 7yrs old, and somehow we have 50k scss o.O)!

We also think it is quite odd, that the "vendor specific prefixes" and the "experimental property" are some how mixed together in the the "experimental-property-usage"! 

Our use-case is e.g. we are well aware that "box-shadow" is experimental, but as long as it is not vendor prefixed, it works for our use case.

So the reason why i am asking, is to understand your opinion and reasoning -> because if it make sense, than there is no reason to change anything, if you say, well we can split this, there is no reason not to, i am happy to contribute and create a pull request! As this is the first time contributing to an Sonarplugin, there are a lot of hurdles in the beginning, so i thought it would be better to ask first, instead of investing a lot of time :D 

Br
Simon

**Comment by racodond** [comment]
Hi Simon,

> Well first of all, nice plugin - pretty useful, and glad there is finally an scss plugin
> We are quite happy with the functionality,

Thanks for your kind words!

> and we are using it, and got really dissapointed in our scss quality (project is 7yrs old, and somehow we have 50k scss o.O)!

:-)

> We also think it is quite odd, that the "vendor specific prefixes" and the "experimental property" are some how mixed together in the the "experimental-property-usage"!

* Vendor-prefix properties are by definition not compatible with all browsers.
* Experimental properties are properties that are defined in some W3C specifications and not implemented yet by all main browsers. That means that not all properties defined in draft W3C specifications are considered experimental. That also means that not all properties defined in CSS2 specifications are considered supported. It really depends on how main browsers handle them.

Either case might lead to unsupported and unexpected behaviors on some browsers. That's why I grouped them under the same coding rule (experimental stuff).

> Our use-case is e.g. we are well aware that "box-shadow" is experimental, but as long as it is not vendor prefixed, it works for our use case.

Regarding `box-shadow`, I'd say that it could now be moved from Experimental to Supported (according to http://caniuse.com/#feat=css-boxshadow for instance). I created the following issue for next version: https://github.com/racodond/sonar-css-plugin/issues/123

To sum up, feel free to ping me if you think that some properties are no longer 'experimental' from your point of view. So that we can assess them and move them from Experimental to Supported.

Tell me if it makes sense to you.

Thanks and have a good day!

David

**Comment by aepfli** [comment]
nice thanks for the quick question! and thank you for moving the boxshadow :D

**Comment by racodond** [comment]
You're welcome! If you want to give it a try, a snapshot is available at https://github.com/racodond/sonar-css-plugin/releases/tag/%23123
